Feature/toasts and error handling

EthanHealy01 opened the pull request:
# Description of Changes

- Added error handling and toast notifications

---

## Checklist

### General

- [ ] I have read the [Contribution Guidelines](https://github.com/Stirling-Tools/Stirling-PDF/blob/main/CONTRIBUTING.md)
- [ ] I have read the [Stirling-PDF Developer Guide](https://github.com/Stirling-Tools/Stirling-PDF/blob/main/devGuide/DeveloperGuide.md) (if applicable)
- [ ] I have read the [How to add new languages to Stirling-PDF](https://github.com/Stirling-Tools/Stirling-PDF/blob/main/devGuide/HowToAddNewLanguage.md) (if applicable)
- [ ] I have performed a self-review of my own code
- [ ] My changes generate no new warnings

### Documentation

- [ ] I have updated relevant docs on [Stirling-PDF's doc repo](https://github.com/Stirling-Tools/Stirling-Tools.github.io/blob/main/docs/) (if functionality has heavily changed)
- [ ] I have read the section [Add New Translation Tags](https://github.com/Stirling-Tools/Stirling-PDF/blob/main/devGuide/HowToAddNewLanguage.md#add-new-translation-tags) (for new translation tags only)

### UI Changes (if applicable)

- [ ] Screenshots or videos demonstrating the UI changes are attached (e.g., as comments or direct attachments in the PR)

### Testing (if applicable)

- [ ] I have tested my changes locally. Refer to the [Testing Guide](https://github.com/Stirling-Tools/Stirling-PDF/blob/main/devGuide/DeveloperGuide.md#6-testing) for more details.

EthanHealy01 committed:
add toasts and error handling

EthanHealy01 committed:
show errors on files in workbench

EthanHealy01 committed:
Merge branch 'V2' into feature/toastsAndErrorHandling

EthanHealy01 committed:
add a dismiss all errors button and change the wording of invalid file errors

EthanHealy01 committed:
Merge branch 'V2' into feature/toastsAndErrorHandling

EthanHealy01 committed:
Merge branch 'feature/toastsAndErrorHandling' of github.com:Stirling-Tools/Stirling-PDF into feature/toastsAndErrorHandling

EthanHealy01 committed:
frontend validation error

EthanHealy01 committed:
add translations and error messages for encrypted/password protected files being edited

EthanHealy01 committed:
linter

ConnorYoh reviewed (changes requested):
http.ts should be apiClient.ts like SaaS. Then our SaaS one can extend it. I think we should do this before merge.

EthanHealy01 committed:
refactor: convert http.ts to axios instance pattern for SaaS compatibility

- Create apiClient via axios.create with baseURL, responseType, withCredentials
- Move response error interceptor to the instance with HMR-safe guards
- Augment instance with CancelToken and isCancel for backward compatibility
- Add named 'api' convenience methods export matching SaaS pattern
- Preserve apiFetch function and existing exports
- Enable SaaS to extend this instance with JWT/401 refresh interceptors

EthanHealy01 committed:
fix merge and tests

EthanHealy01 committed:
add guards
